ML benchmark playbooks

kpouget opened the pull request:
Follow up of #215 (Pam's work) and https://github.com/openshift-psap/ci-artifacts/pull/222 (Coco dataset download example)

kpouget commented:
/test gpu-operator-commit

kpouget commented:
/test gpu-operator-commit

On Fri, Aug 6, 2021 at 4:48 PM openshift-ci[bot] ***@***.***>
wrote:

> @kpouget <https://github.com/kpouget>: The following test *failed*, say
> /retest to rerun all failed tests or /retest-required to rerun all
> mandatory failed tests:
> Test name Commit Details Rerun command
> ci/prow/gpu-operator-commit 20e3f41
> <https://github.com/openshift-psap/ci-artifacts/commit/20e3f418593ca2569cb1fbd24edf970eb20e68bd>
> link
> <https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1423653532141096960> /test
> gpu-operator-commit
>
> Full PR test history
> <https://prow.ci.openshift.org/pr-history?org=openshift-psap&repo=ci-artifacts&pr=235>.
> Your PR dashboard
> <https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:kpouget>.
>
> Instructions for interacting with me using PR comments are available here
> <https://git.k8s.io/community/contributors/guide/pull-requests.md>. If
> you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://go.k8s.io/bot-commands>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift-psap/ci-artifacts/pull/235#issuecomment-894312707>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABZVQIWJKYDK7QXTU4U4PJTT3PY2ZANCNFSM5BVXTEFA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

kpouget commented:
/test gpu-operator-commit

kpouget commented:
@omertuc the mirror seems to reject the certificate:
```
+ md5sum /etc/mirror-secret/mirror.pem
df81f0d8994c7dac9632ddd2a07727fe  /etc/mirror-secret/mirror.pem
+ [[ -e train2017.zip ]]
+ [[ ! -e train2017.zip ]]
+ curl -O -E /etc/mirror-secret/mirror.pem https://mirror-dataset.apps.ci-mirror.psap.aws.rhperfscale.org/coco/train2017.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   153  100   153    0     0    254      0 --:--:-- --:--:-- --:--:--   253

real	0m0.610s
user	0m0.020s
sys	0m0.005s
+ md5sum train2017.zip
7d6137132ff4e08f6848aee8c3ea20e8  train2017.zip
```

locally I get the same:
```
$ curl -E openshift/.creds/entitled-mirror-client-creds.pem https://mirror-dataset.apps.ci-mirror.psap.aws.rhperfscale.org/coco/train2017.zip -s | md5sum
7d6137132ff4e08f6848aee8c3ea20e8  -

$ curl -E openshift/.creds/entitled-mirror-client-creds.pem https://mirror-dataset.apps.ci-mirror.psap.aws.rhperfscale.org/coco/train2017.zip -sSf
curl: (22) The requested URL returned error: 403
```
(I'll update the PR to use `-sSf` for curl ^^)

and I double checked, the PEM is the one from vault:
```
df81f0d8994c7dac9632ddd2a07727fe  /var/run/psap-entitlement-secret/entitled-mirror-client-creds.pem
```
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-psap-ci-artifacts-release-4.9-gpu-operator-e2e-master/1424505781717307392/artifacts/gpu-operator-e2e-master/nightly/build-log.txt

omertuc commented:
> (I'll update the PR to use `-sSf` for curl ^^)

Please use the long form for those flags when you do

> @omertuc the mirror seems to reject the certificate:

Weird, looking into it

omertuc commented:
> > (I'll update the PR to use `-sSf` for curl ^^)
> 
> Please use the long form for those flags when you do
> 
> > @omertuc the mirror seems to reject the certificate:
> 
> Weird, looking into it

Restarting the deployment fixed the issue. Looks like the pod lost access to /mirror_files/ volume mount . Might be some weird Amazon EBS quirk

kpouget commented:
thanks @omertuc , we'll see if that happens periodically or if it was only a one-shot weirdness

/test gpu-operator-commit

kpouget commented:
```
2021-08-10 12:29:11,800 p=60 u=psap-ci-runner n=ansible | <stderr> Error from server: Get "https://10.0.138.191:10250/containerLogs/default/coco-dataset-downloader/coco-dataset-downloader": x509: certificate signed by unknown authority
```
I wonder what that can be, all the log fetching fail because of that, see [gather-extra logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1425069085598683136/artifacts/gpu-operator-commit/gather-extra/build-log.txt)

/test gpu-operator-commit

kpouget commented:
```
+ cd /storage
+ DATASET_FILENAME=train2017.zip
+ [[ -z https://mirror-dataset.apps.ci-mirror.psap.aws.rhperfscale.org/coco ]]
+ [[ ! -e /etc/mirror-secret/mirror.pem ]]
+ md5sum /etc/mirror-secret/mirror.pem
df81f0d8994c7dac9632ddd2a07727fe  /etc/mirror-secret/mirror.pem
+ [[ -e train2017.zip ]]
+ [[ ! -e train2017.zip ]]
+ curl -O -E /etc/mirror-secret/mirror.pem https://mirror-dataset.apps.ci-mirror.psap.aws.rhperfscale.org/coco/train2017.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0 18.0G    0  192k    0     0   162k      0 32:17:32  0:00:01 32:17:31  162k
  0 18.0G    0 7855k    0     0  3709k      0  1:24:51  0:00:02  1:24:49 3707k
  0 18.0G    0 16.7M    0     0  5536k      0  0:56:50  0:00:03  0:56:47 5535k
  0 18.0G    0 27.2M    0     0  6753k      0  0:46:36  0:00:04  0:46:32 6751k
  0 18.0G    0 37.7M    0     0  7595k      0  0:41:26  0:00:05  0:41:21 7774k
  0 18.0G    0 41.1M    0     0  6961k      0  0:45:12  0:00:06  0:45:06 8611k
curl: (18) transfer closed with 19293722744 bytes remaining to read

real	0m6.061s
user	0m0.060s
sys	0m0.103s
```
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1425101673579155456/artifacts/gpu-operator-commit/presubmit-commit/artifacts/000__benchmarking__download_coco_dataset/pod_downloader.log

@omertuc could you check the logs of the mirror to see if there is anything explaining this issue?

omertuc commented:
> ```
> + cd /storage
> + DATASET_FILENAME=train2017.zip
> + [[ -z https://mirror-dataset.apps.ci-mirror.psap.aws.rhperfscale.org/coco ]]
> + [[ ! -e /etc/mirror-secret/mirror.pem ]]
> + md5sum /etc/mirror-secret/mirror.pem
> df81f0d8994c7dac9632ddd2a07727fe  /etc/mirror-secret/mirror.pem
> + [[ -e train2017.zip ]]
> + [[ ! -e train2017.zip ]]
> + curl -O -E /etc/mirror-secret/mirror.pem https://mirror-dataset.apps.ci-mirror.psap.aws.rhperfscale.org/coco/train2017.zip
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
> 
>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
>   0 18.0G    0  192k    0     0   162k      0 32:17:32  0:00:01 32:17:31  162k
>   0 18.0G    0 7855k    0     0  3709k      0  1:24:51  0:00:02  1:24:49 3707k
>   0 18.0G    0 16.7M    0     0  5536k      0  0:56:50  0:00:03  0:56:47 5535k
>   0 18.0G    0 27.2M    0     0  6753k      0  0:46:36  0:00:04  0:46:32 6751k
>   0 18.0G    0 37.7M    0     0  7595k      0  0:41:26  0:00:05  0:41:21 7774k
>   0 18.0G    0 41.1M    0     0  6961k      0  0:45:12  0:00:06  0:45:06 8611k
> curl: (18) transfer closed with 19293722744 bytes remaining to read
> 
> real	0m6.061s
> user	0m0.060s
> sys	0m0.103s
> ```
> 
> [gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1425101673579155456/artifacts/gpu-operator-commit/presubmit-commit/artifacts/000__benchmarking__download_coco_dataset/pod_downloader.log](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1425101673579155456/artifacts/gpu-operator-commit/presubmit-commit/artifacts/000__benchmarking__download_coco_dataset/pod_downloader.log)
> 
> @omertuc could you check the logs of the mirror to see if there is anything explaining this issue?

![image](https://user-images.githubusercontent.com/10882062/128891880-22b1c58e-02c4-4f13-aa5b-f2c3c5323475.png)

omertuc commented:
I'm guessing you'll need to use heavier machinery to download such a large file, something other than curl, or `curl` but with re-tries and picking up a connection where it left off incase it dies abruptly

kpouget commented:
thanks for the logs,

yes, if client and server don't know why the connection was closed, then
we'll need to improve the downloader code ...

weird that it worked at the first try last time, must be luck :#

On Tue, Aug 10, 2021 at 5:11 PM Omer Tuchfeld ***@***.***>
wrote:

> I'm guessing you'll need to use heavier machinery to download such a large
> file, something other than curl, or curl but with re-tries and picking up
> a connection where it left off incase it dies abruptly (if it has such
> capabilities)
>
> EDIT: Something like https://superuser.com/a/142480/177961
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift-psap/ci-artifacts/pull/235#issuecomment-896112555>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABZVQITEWZWLY7JBXKRR5QDT4E6SNANCNFSM5BVXTEFA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

kpouget commented:
/test gpu-operator-commit

kpouget commented:
/test gpu-operator-commit

kpouget commented:
/test gpu-operator-commit

kpouget commented:
[test failed](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1425796205010489344/artifacts/gpu-operator-commit/presubmit-commit/artifacts/000__benchmarking__download_coco_dataset/pod_downloader.log) because only 890MB of data was downloaded before reaching the retry count. Increasing it to 20min



/test gpu-operator-commit

kpouget commented:
/test gpu-operator-commit

On Mon, Aug 23, 2021 at 12:21 PM openshift-ci[bot] ***@***.***>
wrote:

> @kpouget <https://github.com/kpouget>: The following test *failed*, say
> /retest to rerun all failed tests or /retest-required to rerun all
> mandatory failed tests:
> Test name Commit Details Rerun command
> ci/prow/gpu-operator-commit 245a3f0
> <https://github.com/openshift-psap/ci-artifacts/commit/245a3f02a56eb14c469a37b309fb44ab4a29e599>
> link
> <https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1429746816731058176> /test
> gpu-operator-commit
>
> Full PR test history
> <https://prow.ci.openshift.org/pr-history?org=openshift-psap&repo=ci-artifacts&pr=235>.
> Your PR dashboard
> <https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:kpouget>.
>
> Instructions for interacting with me using PR comments are available here
> <https://git.k8s.io/community/contributors/guide/pull-requests.md>. If
> you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://go.k8s.io/bot-commands>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift-psap/ci-artifacts/pull/235#issuecomment-903638387>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABZVQIX2IWQJDFXJGXNV4BDT6IOLHANCNFSM5BVXTEFA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

kpouget commented:
/test gpu-operator-commit

On Mon, Aug 23, 2021 at 1:20 PM openshift-ci[bot] ***@***.***>
wrote:

> @kpouget <https://github.com/kpouget>: The following test *failed*, say
> /retest to rerun all failed tests or /retest-required to rerun all
> mandatory failed tests:
> Test name Commit Details Rerun command
> ci/prow/gpu-operator-commit 431644c
> <https://github.com/openshift-psap/ci-artifacts/commit/431644cfce2da808acd987753775eafc4b802947>
> link
> <https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1429761325461934080> /test
> gpu-operator-commit
>
> Full PR test history
> <https://prow.ci.openshift.org/pr-history?org=openshift-psap&repo=ci-artifacts&pr=235>.
> Your PR dashboard
> <https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:kpouget>.
>
> Instructions for interacting with me using PR comments are available here
> <https://git.k8s.io/community/contributors/guide/pull-requests.md>. If
> you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://go.k8s.io/bot-commands>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift-psap/ci-artifacts/pull/235#issuecomment-903676472>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABZVQIXPKSV7DNDKA7LUCKTT6IVHJANCNFSM5BVXTEFA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

kpouget commented:
/test gpu-operator-commit

kpouget commented:
/test gpu-operator-commit

On Mon, Aug 30, 2021 at 11:59 AM openshift-ci[bot] ***@***.***>
wrote:

> @kpouget <https://github.com/kpouget>: The following test *failed*, say
> /retest to rerun all failed tests or /retest-required to rerun all
> mandatory failed tests:
> Test name Commit Details Rerun command
> ci/prow/gpu-operator-commit dca06c7
> <https://github.com/openshift-psap/ci-artifacts/commit/dca06c760f07effec690a5cdffef837252ef1505>
> link
> <https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1432277895220301824> /test
> gpu-operator-commit
>
> Full PR test history
> <https://prow.ci.openshift.org/pr-history?org=openshift-psap&repo=ci-artifacts&pr=235>.
> Your PR dashboard
> <https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:kpouget>.
>
> Instructions for interacting with me using PR comments are available here
> <https://git.k8s.io/community/contributors/guide/pull-requests.md>. If
> you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://go.k8s.io/bot-commands>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift-psap/ci-artifacts/pull/235#issuecomment-908209498>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABZVQIUKXA3XNMTCRH4RAY3T7NI7NANCNFSM5BVXTEFA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

kpouget commented:
/test gpu-operator-commit

On Mon, Aug 30, 2021 at 12:20 PM openshift-ci[bot] ***@***.***>
wrote:

> @kpouget <https://github.com/kpouget>: The following test *failed*, say
> /retest to rerun all failed tests or /retest-required to rerun all
> mandatory failed tests:
> Test name Commit Details Rerun command
> ci/prow/gpu-operator-commit f1b03a6
> <https://github.com/openshift-psap/ci-artifacts/commit/f1b03a60ee20db7ddd50cf11ef879c337a6d3b7e>
> link
> <https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1432283153967353856> /test
> gpu-operator-commit
>
> Full PR test history
> <https://prow.ci.openshift.org/pr-history?org=openshift-psap&repo=ci-artifacts&pr=235>.
> Your PR dashboard
> <https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:kpouget>.
>
> Instructions for interacting with me using PR comments are available here
> <https://git.k8s.io/community/contributors/guide/pull-requests.md>. If
> you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://go.k8s.io/bot-commands>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift-psap/ci-artifacts/pull/235#issuecomment-908225321>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABZVQIS5RKAY5WODIPRPDEDT7NLQJANCNFSM5BVXTEFA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

kpouget commented:
/test gpu-operator-commit

kpouget commented:
/test gpu-operator-commit

kpouget commented:
/test gpu-operator-commit

kpouget commented:
/test gpu-operator-commit

kpouget commented:
/test gpu-operator-commit

On Tue, Aug 31, 2021 at 11:42 AM openshift-ci[bot] ***@***.***>
wrote:

> @kpouget <https://github.com/kpouget>: The following test *failed*, say
> /retest to rerun all failed tests or /retest-required to rerun all
> mandatory failed tests:
> Test name Commit Details Rerun command
> ci/prow/gpu-operator-commit 2b83913
> <https://github.com/openshift-psap/ci-artifacts/commit/2b83913a7e108da72b6868a92e37afed8e93ecee>
> link
> <https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1432634473513488384> /test
> gpu-operator-commit
>
> Full PR test history
> <https://prow.ci.openshift.org/pr-history?org=openshift-psap&repo=ci-artifacts&pr=235>.
> Your PR dashboard
> <https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:kpouget>.
>
> Instructions for interacting with me using PR comments are available here
> <https://git.k8s.io/community/contributors/guide/pull-requests.md>. If
> you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://go.k8s.io/bot-commands>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift-psap/ci-artifacts/pull/235#issuecomment-909074540>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABZVQIWGBQXTV2ONIBMKPDTT7SPZBANCNFSM5BVXTEFA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

kpouget commented:
```
            "message": "0/7 nodes are available: 1 node(s) had volume node affinity conflict, 3 Insufficient nvidia.com/gpu, 3 node(s) had taint {node-role.kubernetes.io/master: }, that the pod didn't tolerate.",
```

kpouget commented:
/test gpu-operator-commit

kpouget commented:
/test gpu-operator-e2e

kpouget commented:
/test gpu-operator-commit

kpouget commented:
/test gpu-operator-commit

On Wed, Sep 1, 2021 at 5:42 PM openshift-ci[bot] ***@***.***>
wrote:

> @kpouget <https://github.com/kpouget>: The following test *failed*, say
> /retest to rerun all failed tests or /retest-required to rerun all
> mandatory failed tests:
> Test name Commit Details Rerun command
> ci/prow/gpu-operator-commit 6fb175b
> <https://github.com/openshift-psap/ci-artifacts/commit/6fb175be746a5dab7b99980054556a7cb508fe8d>
> link
> <https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1433084622937264128> /test
> gpu-operator-commit
>
> Full PR test history
> <https://prow.ci.openshift.org/pr-history?org=openshift-psap&repo=ci-artifacts&pr=235>.
> Your PR dashboard
> <https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:kpouget>.
>
> Instructions for interacting with me using PR comments are available here
> <https://git.k8s.io/community/contributors/guide/pull-requests.md>. If
> you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://go.k8s.io/bot-commands>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift-psap/ci-artifacts/pull/235#issuecomment-910408516>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABZVQIRSJJKMBWVG4KBME33T7ZCVLANCNFSM5BVXTEFA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

kpouget commented:
/test gpu-operator-commit

kpouget commented:
/test gpu-operator-commit

On Fri, Sep 3, 2021 at 4:31 PM openshift-ci[bot] ***@***.***>
wrote:

> @kpouget <https://github.com/kpouget>: The following test *failed*, say
> /retest to rerun all failed tests or /retest-required to rerun all
> mandatory failed tests:
> Test name Commit Details Rerun command
> ci/prow/gpu-operator-commit dd0bd23
> <https://github.com/openshift-psap/ci-artifacts/commit/dd0bd23052dd7867869c01ccde5bc1ea6b6e439b>
> link
> <https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1433786908319682560> /test
> gpu-operator-commit
>
> Full PR test history
> <https://prow.ci.openshift.org/pr-history?org=openshift-psap&repo=ci-artifacts&pr=235>.
> Your PR dashboard
> <https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:kpouget>.
>
> Instructions for interacting with me using PR comments are available here
> <https://git.k8s.io/community/contributors/guide/pull-requests.md>. If
> you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://go.k8s.io/bot-commands>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift-psap/ci-artifacts/pull/235#issuecomment-912583822>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABZVQIWKPFT54HXABSIXGGTUADL3XANCNFSM5BVXTEFA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

kpouget commented:
/test gpu-operator-commit

On Fri, Sep 3, 2021 at 7:32 PM openshift-ci[bot] ***@***.***>
wrote:

> @kpouget <https://github.com/kpouget>: The following test *failed*, say
> /retest to rerun all failed tests or /retest-required to rerun all
> mandatory failed tests:
> Test name Commit Details Rerun command
> ci/prow/gpu-operator-commit b8e753e
> <https://github.com/openshift-psap/ci-artifacts/commit/b8e753e2996c995896a1a32cebf90bab5059c276>
> link
> <https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1433832343289729024> /test
> gpu-operator-commit
>
> Full PR test history
> <https://prow.ci.openshift.org/pr-history?org=openshift-psap&repo=ci-artifacts&pr=235>.
> Your PR dashboard
> <https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:kpouget>.
>
> Instructions for interacting with me using PR comments are available here
> <https://git.k8s.io/community/contributors/guide/pull-requests.md>. If
> you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://go.k8s.io/bot-commands>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift-psap/ci-artifacts/pull/235#issuecomment-912697938>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABZVQIXBGYVEN3TLN7QD5CLUAEBDTANCNFSM5BVXTEFA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

kpouget commented:
/test gpu-operator-commit

On Fri, Sep 3, 2021 at 9:33 PM openshift-ci[bot] ***@***.***>
wrote:

> @kpouget <https://github.com/kpouget>: The following test *failed*, say
> /retest to rerun all failed tests or /retest-required to rerun all
> mandatory failed tests:
> Test name Commit Details Rerun command
> ci/prow/gpu-operator-commit b8e753e
> <https://github.com/openshift-psap/ci-artifacts/commit/b8e753e2996c995896a1a32cebf90bab5059c276>
> link
> <https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1433854060120772608> /test
> gpu-operator-commit
>
> Full PR test history
> <https://prow.ci.openshift.org/pr-history?org=openshift-psap&repo=ci-artifacts&pr=235>.
> Your PR dashboard
> <https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:kpouget>.
>
> Instructions for interacting with me using PR comments are available here
> <https://git.k8s.io/community/contributors/guide/pull-requests.md>. If
> you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://go.k8s.io/bot-commands>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift-psap/ci-artifacts/pull/235#issuecomment-912764944>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABZVQIRBPBOESCJX45DLGYDUAEPH3ANCNFSM5BVXTEFA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

kpouget commented:
/test gpu-operator-commit

kpouget commented:
\o/
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1434121308366442496/artifacts/gpu-operator-commit/presubmit-commit/artifacts/013__benchmarking__run_nvidiadl_ssd/nvidiadl_ssd_pod.log
@omertuc @ArangoGutierrez @dagrayvid PR is ready for review, let me know if you have any comment

\hold <-- there's still a `WIP` commit for triggering the benchmark

kpouget commented:
/test gpu-operator-commit

On Fri, Sep 3, 2021 at 11:01 PM openshift-ci[bot] ***@***.***>
wrote:

> @kpouget <https://github.com/kpouget>: The following test *failed*, say
> /retest to rerun all failed tests or /retest-required to rerun all
> mandatory failed tests:
> Test name Commit Details Rerun command
> ci/prow/gpu-operator-commit 1db155c
> <https://github.com/openshift-psap/ci-artifacts/commit/1db155ccc7e865469d95ef8044837928d1ce2d33>
> link
> <https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1433877825961594880> /test
> gpu-operator-commit
>
> Full PR test history
> <https://prow.ci.openshift.org/pr-history?org=openshift-psap&repo=ci-artifacts&pr=235>.
> Your PR dashboard
> <https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:kpouget>.
>
> Instructions for interacting with me using PR comments are available here
> <https://git.k8s.io/community/contributors/guide/pull-requests.md>. If
> you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://go.k8s.io/bot-commands>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift-psap/ci-artifacts/pull/235#issuecomment-912807995>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABZVQIVOFB2ID7COEDFSXBDUAEZSLANCNFSM5BVXTEFA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

omertuc commented on roles/benchmarking_deploy_coco_dataset/tasks/main.yml:
```diff
@@ -0,0 +1,84 @@
+- name: Ensure that a valid hostname name was passed in parameter
+  command: oc get nodes -l kubernetes.io/hostname={{ benchmarking_node_hostname }} -oname
+
+- name: Delete the mirror secret, if it exists
```
Consider a more idempotent approach, or protect this command with an `--allow-overwrite` toolbox flag

omertuc replied:
Oh sorry, you're just deleting the secret, not the mirror. Ignore me

kpouget replied:
hum, you though it was the PVC that was being deleted I guess ;-)
no, no command doing that for now ;-)

the test is actually validating that the command is idempotent, and doesn't  download twice the dataset (and no more md5sum during the 2nd try, only a simple flag-file check ;-))
```
for i in 1 2; do
    ./run_toolbox.py benchmarking download_coco_dataset \
                     "$gpu_node_hostname" \
                     --mirror_base_url=https://mirror-dataset.apps.ci-mirror.psap.aws.rhperfscale.org/coco \
                     --client-cert=${PSAP_SECRET_PATH}/entitled-mirror-client-creds.pem
done
```

kpouget commented:
/hold

I'll reorder the commit and get the test commited-and-reverted with https://github.com/openshift-psap/ci-artifacts/pull/251

kpouget commented:
/test gpu-operator-commit

kpouget commented:
Hello @omertuc @dagrayvid can get merge this PR?

The tests show that:
1. the coco dataset is properly [downloaded](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1435263283366989824/artifacts/gpu-operator-commit/presubmit-commit/artifacts/011__benchmarking__download_coco_dataset/) from the CI mirror
2. calling this command twice has [no side effect](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1435263283366989824/artifacts/gpu-operator-commit/presubmit-commit/artifacts/012__benchmarking__download_coco_dataset/pod_downloader.log), the dataset is not re-downloaded, and checking it takes only [1min](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1435263283366989824/artifacts/gpu-operator-commit/presubmit-commit/artifacts/012__benchmarking__download_coco_dataset/_ansible.log) as I use a file-flag `$FILENAME.extracted` to ensure that the file was correctly downloaded (`md5sum` checked) and extracted. This file-flag avoid computing the md5sum of big files that takes several minutes to complete
3. the SSD benchmark is executed (8min), its Pod logs are [stored](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1435263283366989824/artifacts/gpu-operator-commit/presubmit-commit/artifacts/013__benchmarking__run_nvidiadl_ssd/nvidiadl_ssd_pod.log),  as well as the execution [status](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1435263283366989824/artifacts/gpu-operator-commit/presubmit-commit/artifacts/013__benchmarking__run_nvidiadl_ssd/status), in order to simplify post-processing in the future.

Overall, what is missing here is that the benchmark pod image is [hard-coded](https://github.com/openshift-psap/ci-artifacts/pull/235/files#diff-5c81f9296951e90942fc06aed989a0c7f435b53f5d4025680a5c4105ebd24637R12) (`quay.io/openshift-psap/nvidiadl-ssd-training-benchmark`) and was manually generated, but this will be solved later on.

Note that the 8-minute execution time includes the image downloading (4.5GB).

kpouget commented:
> Hello @omertuc @dagrayvid can get merge this PR?
> 
> The tests show that:
> 
>     1. the coco dataset is properly [downloaded](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1435263283366989824/artifacts/gpu-operator-commit/presubmit-commit/artifacts/011__benchmarking__download_coco_dataset/) from the CI mirror
> 
>     2. calling this command twice has [no side effect](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1435263283366989824/artifacts/gpu-operator-commit/presubmit-commit/artifacts/012__benchmarking__download_coco_dataset/pod_downloader.log), the dataset is not re-downloaded, and checking it takes only [1min](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1435263283366989824/artifacts/gpu-operator-commit/presubmit-commit/artifacts/012__benchmarking__download_coco_dataset/_ansible.log) as I use a file-flag `$FILENAME.extracted` to ensure that the file was correctly downloaded (`md5sum` checked) and extracted. This file-flag avoid computing the md5sum of big files that takes several minutes to complete
> 
>     3. the SSD benchmark is executed (8min), its Pod logs are [stored](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1435263283366989824/artifacts/gpu-operator-commit/presubmit-commit/artifacts/013__benchmarking__run_nvidiadl_ssd/nvidiadl_ssd_pod.log),  as well as the execution [status](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1435263283366989824/artifacts/gpu-operator-commit/presubmit-commit/artifacts/013__benchmarking__run_nvidiadl_ssd/status), in order to simplify post-processing in the future.
> 
> 
> Overall, what is missing here is that the benchmark pod image is [hard-coded](https://github.com/openshift-psap/ci-artifacts/pull/235/files#diff-5c81f9296951e90942fc06aed989a0c7f435b53f5d4025680a5c4105ebd24637R12) (`quay.io/openshift-psap/nvidiadl-ssd-training-benchmark`) and was manually generated, but this will be solved later on.
> 
> Note that the 8-minute execution time includes the image downloading (4.5GB).

Hello @jmencak, would you have a moment to have a look at this PR?
it's not complete yet as it's not included in any nightly run, but that will be follow-up work for someone else!

jmencak commented on roles/benchmarking_deploy_coco_dataset/files/entrypoint.sh:
```diff
@@ -0,0 +1,87 @@
+fi
+
+echo "Installing 'unzip' ..."
+dnf install unzip -y --quiet
```
Do we also want to `exit 1` on failure of installing `unzip`?

jmencak commented on roles/benchmarking_deploy_coco_dataset/files/entrypoint.sh:
```diff
@@ -0,0 +1,87 @@
+    grep "  $dataset_filename" checksum.md5sum | sed "s/$dataset_filename/-/" > stdin.md5sum
+
+    if [[ -e "$dataset_filename" ]]; then
+        echo "INFO: Found an existing '$dataset_filename'. Checking its md5sum ..."
```
A future PR should probably cleanup all these `echo "INFO/FATAL: " into simple functions such as info/fatal()` and also probably log to `STDERR` vs. `STDOUT`.

jmencak commented on roles/benchmarking_deploy_coco_dataset/meta/main.yml:
```diff
@@ -0,0 +1,3 @@
+---
+dependencies:
+  - role: check_deps
```
Github helps you to give you a little nudge here.  Most files with no EOL are a no-no in my view.

jmencak commented on roles/benchmarking_run_nvidiadl_ssd/tasks/main.yml:
```diff
@@ -0,0 +1,64 @@
+    oc get pvc/{{ benchmarking_coco_dataset_pvc_name }}
+       -n {{ benchmarking_namespace }}
+
+- name: Delete the entrypoint ConfigMap, it it exists
```
`s/it it/if it/`

kpouget replied:
+1

jmencak commented on roles/benchmarking_run_nvidiadl_ssd/tasks/main.yml:
```diff
@@ -0,0 +1,64 @@
+
+- name: Store the status of the benchmark execution (for post-processing)
+  shell:
+    echo "{{  wait_benchmark_pod_cmd.stdout }}" > "{{ artifact_extra_logs_dir }}/status"
```
`s/  / /` (double space vs. single)

kpouget replied:
+1
should have been caught by `ansible-lint` again :/ :/

jmencak commented on roles/benchmarking_run_nvidiadl_ssd/templates/nvidiadl-ssd-pod.yml.j2:
```diff
@@ -0,0 +1,36 @@
+spec:
+  restartPolicy: Never
+  nodeSelector:
+    kubernetes.io/hostname: {{ benchmarking_node_hostname}}
```
`s/e}/e }/`

kpouget replied:
damn copy and paste :#

jmencak commented on roles/benchmarking_deploy_coco_dataset/templates/pod.yml.j2:
```diff
@@ -0,0 +1,47 @@
+spec:
+  restartPolicy: Never
+  nodeSelector:
+    kubernetes.io/hostname: {{ benchmarking_node_hostname}}
```
`s/e}/e }/`

kpouget replied:
+1
this one should have been caught by `ansible-lint` .... :/

jmencak reviewed (commented):
It is hard for me to understand the PR fully without testing and playing with it a bit.  Nevertheless, the tests pass the code seems sane.  Only nits.  Once addressed, I'm happy to approve.

kpouget commented on roles/benchmarking_deploy_coco_dataset/defaults/main/config.yml:
```diff
@@ -0,0 +1,16 @@
+
+# Base URL where the coco dataset files should be fetched
+# if empty, download from the upstream repository
+benchmarking_coco_dataset_mirror_base_url:
```
good catch, I don't think it changes anything. Will fix it for consistency

kpouget reviewed (commented):
thanks @jmencak for the review, all the comments should have been fixed with the last push

/hold <-- added back the TEST route path to validate the last changes + rebase

kpouget commented:
/test gpu-operator-commit

jmencak commented:
/lgtm
/hold
Please cancel the hold once you're happy with the testing, Kevin.

kpouget commented:
> /lgtm
> /hold
> Please cancel the hold once you're happy with the testing, Kevin.

thanks a lot for the review @jmencak :)

--- 

test failed after rebase, the GPU Operator now needs `--namespace openshift-operators` to be deployed.

testing on 4.6 as 4.7 and 4.8 seems to be affected by the missing kernel package, so the driver cannot be built.

/test 4.6-gpu-operator-e2e

jmencak commented on build/root/usr/local/bin/ci_entrypoint_custom.sh:
```diff
@@ -1,11 +1,34 @@
+
+if [ -z "$gpu_node_hostname" ]; then
+    echo "Couldn't find the GPU node ..."
+    oc get nodes --show-labels | sed 's|,|,- |g' | tr ',' '\n'
```
The output will be messy on SNO clusters for example, where you have dual (master,worker) roles separated with `','`.  If that's not a problem to you, happy to `/lgtm` again.

kpouget replied:
good point :)
this file `ci_entrypoint_custom.sh` will be reverted once the PR is merged, so not an issue here

kpouget commented:
>  ci/prow/4.6-gpu-operator-e2e — Job failed.                    
```
* could not run steps: step gpu-operator-e2e failed: failed to wait for the created cluster claim to become ready: timed out waiting for the condition 
```
:( 

trying again, but maybe we're requestiong 4.6 clusters from a non-existing pool (or pool is just empty, we'll see)

/test 4.6-gpu-operator-e2e

kpouget commented:
I'll wait for this PR to be merged: https://github.com/openshift-psap/ci-artifacts/pull/255
it will offer a way to easily add and trigger custom testing, similarly to what this PR is doing on its top commit

kpouget commented:
/test custom

kpouget commented:
/test custom

kpouget commented:
/test custom

kpouget commented:
the ci-mirror server wasn't feeling very well (high memory pressure) ...

should be better now, hopefully

/test custom

On Fri, Sep 17, 2021 at 4:55 PM openshift-ci[bot] ***@***.***>
wrote:

> @kpouget <https://github.com/kpouget>: The following tests *failed*, say
> /retest to rerun all failed tests or /retest-required to rerun all
> mandatory failed tests:
> Test name Commit Details Rerun command
> ci/prow/gpu-operator-commit dfd3b7b
> <https://github.com/openshift-psap/ci-artifacts/commit/dfd3b7b2c650dfbd20cf03fae8059977adb87d81>
> link
> <https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-commit/1436240395926245376> /test
> gpu-operator-commit
> ci/prow/4.6-gpu-operator-e2e 3b50568
> <https://github.com/openshift-psap/ci-artifacts/commit/3b5056865f81cbd0e01af0316e3ea8f162cc84dc>
> link
> <https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-4.6-gpu-operator-e2e/1436287129545084928> /test
> 4.6-gpu-operator-e2e
> ci/prow/custom f72b7dd
> <https://github.com/openshift-psap/ci-artifacts/commit/f72b7dd5749827fa01b915e66633f2eabb9a7de2>
> link
> <https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/235/pull-ci-openshift-psap-ci-artifacts-master-custom/1438868506341806080> /test
> custom
>
> Full PR test history
> <https://prow.ci.openshift.org/pr-history?org=openshift-psap&repo=ci-artifacts&pr=235>.
> Your PR dashboard
> <https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:kpouget>.
>
> Instructions for interacting with me using PR comments are available here
> <https://git.k8s.io/community/contributors/guide/pull-requests.md>. If
> you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://go.k8s.io/bot-commands>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift-psap/ci-artifacts/pull/235#issuecomment-921862933>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABZVQIQLB3J4TRJHGIMUN3DUCNJE3ANCNFSM5BVXTEFA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

kpouget commented:
> /lgtm
> /hold
> Please cancel the hold once you're happy with the testing, Kevin.

@jmencak PR is now good for merging, 
the only think that changed since your previous `/lgtm` is the location of the test script `[testing/custom/run_cocodataset_nvidiadl_ssd.sh](https://github.com/openshift-psap/ci-artifacts/blob/f72b7dd5749827fa01b915e66633f2eabb9a7de2/testing/custom/run_cocodataset_nvidiadl_ssd.sh)` which now belongs to `testing/custom` and will stay there for future reference & testing

kpouget commented:
/approve

kpouget commented:
/hold cancel

jmencak commented on roles/benchmarking_deploy_coco_dataset/tasks/main.yml:
```diff
@@ -0,0 +1,84 @@
+  command:
+    oc delete secret/mirror-secret
+       -n {{ benchmarking_namespace }}
+       --ignore-not-found=true
```
`--ignore-not-found` is true by default, no need for `--ignore-not-found=true`
Similarly below.

kpouget replied:
oh you mean `--ignore-not-found=true` is like `test == true`?
true indeed :) but there are 46 places where it should be updated :#
will be for another PR!

jmencak commented on roles/benchmarking_deploy_coco_dataset/tasks/main.yml:
```diff
@@ -0,0 +1,84 @@
+    oc logs pod/coco-dataset-downloader
+       -n {{ benchmarking_namespace }}
+       > {{ artifact_extra_logs_dir }}/pod_downloader.log
+
```
Extra new line.

jmencak reviewed (approved):
/lgtm
Thank you Kevin, looks good, only nits that can be fixed later on.

kpouget commented:
> /lgtm
> Thank you Kevin, looks good, only nits that can be fixed later on.

thanks @jmencak :)

kpouget commented:
/override ci/prow/gpu-operator-e2e
